src/Integration/WPCOMProxy.php

```diff
@@ -0,0 +1,209 @@
+<?php
+declare( strict_types=1 );
+
+namespace Automattic\WooCommerce\GoogleListingsAndAds\Integration;
+
+use Automattic\WooCommerce\GoogleListingsAndAds\Infrastructure\Registerable;
+use Automattic\WooCommerce\GoogleListingsAndAds\Infrastructure\Service;
+use WP_REST_Response;
+use WP_REST_Request;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Class WPCOMProxy
+ *
+ * Initializes the hooks to filter the data sent to the WPCOM proxy depending on the query parameter gla_syncable.
+ *
+ * @package Automattic\WooCommerce\GoogleListingsAndAds\Integration
+ */
+class WPCOMProxy implements Service, Registerable {
+
+	/**
+	 * The Post types to be filtered.
+	 *
+	 * @var array
+	 */
+	public static $post_types_to_be_filter = [
+		'product'           => [
+			'meta_query' => [
+				[
+					'key'     => '_wc_gla_visibility',
+					'value'   => 'sync-and-show',
+					'compare' => '=',
+				],
+			],
+		],
+		'shop_coupon'       => [
+			'meta_query' => [
+				[
+					'key'     => '_wc_gla_visibility',
+					'value'   => 'sync-and-show',
+					'compare' => '=',
+				],
+				[
+					'key'     => 'customer_email',
+					'compare' => 'NOT EXISTS',
+				],
+			],
+		],
+		'product_variation' => [
+			'meta_query' => null,
+		],
+	];
+
+	/**
+	 * Register all filters.
+	 */
+	public function register(): void {
+		// Allow to filter by gla_syncable.
+		add_filter(
+			'woocommerce_rest_query_vars',
+			function ( $valid_vars ) {
+				$valid_vars[] = 'gla_syncable';
+				return $valid_vars;
+			}
+		);
+
+		foreach ( array_keys( self::$post_types_to_be_filter ) as $object_type ) {
+			$this->register_object_types_filter( $object_type );
+		}
+	}
+
+	/**
+	 * Register the filters for a specific object type.
+	 *
+	 * @param string $object_type The object type.
+	 */
+	protected function register_object_types_filter( string $object_type ): void {
+		add_filter( 'woocommerce_rest_prepare_' . $object_type . '_object', [ $this, 'check_item_is_syncable' ], 9, 3 );
+
+		add_filter(
+			'woocommerce_rest_prepare_' . $object_type . '_object',
+			[ $this, 'filter_metadata' ],
+			10,
+			3
+		);
+
+		add_filter(
+			'woocommerce_rest_' . $object_type . '_object_query',
+			[ $this, 'filter_by_metaquery' ],
+			10,
+			2
+		);
+	}
+
+	/**
+	 * Whether the data should be filtered.
+	 *
+	 * @param WP_REST_Request $request The request object.
+	 *
+	 * @return bool
+	 */
+	protected function should_filter_data( WP_REST_Request $request ): bool {
+		return $request->get_param( 'gla_syncable' ) !== null;
```

Comment:
I appreciate this won't be widely used, however, I'm curious why we're only relying on the presence of the parameter and not paying attention to the value?

If we send a request to both:

- `/wp-json/wc/v3/products?gla_syncable=true`
- `/wp-json/wc/v3/products?gla_syncable=false`

I would expect different results but at the moment we're going to get the same filtered results for both.

Reply:
Initially, my intention was only to check the presence because we include the parameter only for the GL&A app, making the value irrelevant. However, I understand your point that you would expect different results with the above requests so I adjusted the code. I didn't use "true" or "false" because in WPCOM the boolean gets sent as "1" or "0"